projects/_posts/2016-08-07-Mapping.md

```diff
@@ -0,0 +1,268 @@
+
+{% include x3d-model.html caption="Example Phase II result + MeshLab surface reconstruction" source="/images/projects/2016-08-07-Mapping/face-reconstruction.x3d" %}
+
+## Phase I
+
+Goals:
+
+1. Hardware selection
+-  CruizCore XG1300L gyroscope for low bias drift and great accuracy
+-  XV11 Lidar for unmatched... price 
+2. Drivers
+-  CruizCore XG1300L driver on top of wonderful @dlech sensor driver framework
+-  XV11 Lidar in userspace using extensive information from [xv11hacking]
+3. Correct geometry
+-  nobody (on the internet) seems to treat XV11 [rotational geometry] correctly
+4. LIDAR standard deviation
+-  evaluation with respect to distance
+5. Experimental 2D mapping
+-  experimental code to learn and throw away
+
+The Phase I is finished with results:
+
+-   [Using the XV11 Lidar tutorial]
+-   [xv11lidar] C/C++ library
+-   [mi-xg1300l] CruizCore XG1300L driver 
+-   [ev3dev-mapping-abandoned] repository
+
+The Phase I robot in action:
+
+-   [How to interface XV11 LIDAR to EV3 using ev3dev]
+-   [EV3 Gyro vs CruizCore XG1300L vs Odometry - Position Estimation]
+
+{% include screenshot.html source="/images/projects/2016-08-07-Mapping/robot-phaseI.jpg" caption="Hardware used for testing in Phase I" %}
+
+## Phase II
+
+Goals:
+
+1. Cross-platform visualisation/computation/control
+-   [Unity] chosen for amazing [platforms] support
+2. Modular architecture design
+-   modules as simple command line programs in any language
+-   supervisor on top that does 'magic'
+-   UI components counterpart of modules 
+3. 3D mapping with lidars in arbitrary position/rotation
+-   XY and XZ plane support with any position, generalizing is easy
+4. Mapping in movement
+-   precise per reading interpolation for position/heading
+5. Support for hundreds of thousands of points on average laptop
+-   lidar spits around 100k points a minute!
+6. Network communication recording/replaying
+-   for running same data with different algorithms/parameters added features
+-   for reprocessing the data on more powerful machine if needed
+-   for near-perfect path repeating
+7. Data export to widely-accepted file format
+-   [ply](https://en.wikipedia.org/wiki/PLY_(file_format)) chosen for ease of implementation
+8. Multiple lidars support at the same time
+-   just use multiple Laser components
+9. Multiple robots support at the same time
+-   no problem but drive (keyboard, pad) scripted for single robot now
+10. Extensibility for different hardware
+-   this needs writing own modules/components
+
+Phase II is functionally finished now with results:
+
+-    [ev3dev-mapping-ui] Unity project repository
+-    [ev3dev-mapping-modules] robot modules repository
+-    [ev3dev-mapping-ui-udp] recorded UDP communication from some mappings
+-    [ev3dev-mapping-results] with screenshots/3D models
+
+The Phase II robot in action:
+
+-    [3D mapping/scanning project with ev3dev OS and Unity UI]
+
+## State of the Project 
+
+Information from 10.08.2016.
```

Comment:
Writing the date as "10 Aug. 2016" might make it a bit more international friendly (at least for us backwards Americans).